### PR TITLE
Add Observable instance and static methods

### PR DESCRIPTION
**Pull request by bfricka**
This adds support for all instance and static methods (operators).

I'd also like to open a separate PR to get this building w/ RxJS latest stable (5.0.3), but they've changed it so built docs aren't part of the repo, so I'll have to add a couple steps and refactor a bit. Let me know if you're amenable.

Edit: Had to change up the RegExp a bit b/c not all elements follow that exact pattern (some have extraneous stuff), which you can see: `Array.from(document.querySelectorAll('[data-ice="anchor"]')).forEach(el => console.log(el.attributes.length))`

### CODE REVIEW

**Comment by royling** [comment]
@bfricka thanks for the PR, that's a good idea to include static/instance methods as well.

However, to make this general, it'd be better list methods of a class as [__table of contents support__](https://kapeli.com/docsets#tableofcontents) instead.

Your thoughts?

**Comment by bfricka** [comment]
I agree that would be a nice feature, and I'll probably implement it when I get some time, but I don't think it should necessarily be a requirement for merging this PR. After all, it's just a convenience for browsing through the docs, whereas operators are the fundamental building blocks of composition in RxJS, and not having them indexed really limits the usefulness of this docset (which, of course, is why I did this).

My priorities are going to be:

1. This (done, unless you have some comments on the implementation)
2. Get a build running off the stable tags so docs can remain up to date
3. TOC

**Comment by royling** [comment]
>operators are the fundamental building blocks of composition in RxJS, and not having them indexed really limits the usefulness of this docset

Totally agree, but will we keep these operators when TOC would be implemented? or delete them then?

File: index.js [hunk 1]
```diff
@@ -4,10 +4,14 @@ const Sequelize = require('sequelize');
...
 
+const obsPath = 'rxjs/class/es6/Observable.js~Observable.html';
+const obsPattern = /^\s*<h3 data-ice="anchor" id="((?:static|instance)-method-(\S*))"[\s\S]*$/i;
+
 const typeMap = {
     class: 'Class',
     interface: 'Interface',
-    typedef: 'Type'
+    typedef: 'Type',
+    method: 'Method'
```
Comment (royling) [inline index.js]: I suggest to make entry type explicitly as 'Operator'

File: index.js [hunk 2]
```diff
@@ -27,6 +31,23 @@ rl.on('line', line => {
...
+    let matches = obsPattern.exec(line);
+    if (matches && matches.length === 3) {
+        objects.push({
+            type: typeMap.method,
+            path: `${obsPath}#${matches[1]}`,
+            name: matches[2]
+        });
+    }
+});
+
```
Comment (royling) [inline index.js:50]: Let's add a TODO here for refactoring later, repeated code here 😿

**Review by royling — APPROVED** [review]

**Pull request merged**